### PR TITLE
feat(`server`): extract parseTRPCMessage for 3rd party modules

### PR DESCRIPTION
**Pull request by Guria**
## 🎯 Changes

This PR adds an adapter/link pair to run trpc server in a webworker.
It might be useful for example as temporary solution for migrating from SPA without nodejs backend.

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [ ] If necessary, I have added documentation related to the changes made. 
- [ ] I have added or updated the tests related to the changes made.

Docs and tests after discussion :)

### CODE REVIEW

**Review by KATT — CHANGES_REQUESTED** [review]
We won't add more adapters at the moment - please make a separate PR where you expose the things you need to realise this

**Comment by Guria** [comment]
@KATT updated

**Comment by kamilogorek** [comment]
Change looks good to me, yet some ws tests are failing. Can you investigate?

File: packages/server/src/rpc/parseTRPCMessage.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+/* istanbul ignore next -- @preserve */
+function assertIsJSONRPC2OrUndefined(
+  obj: unknown,
+): asserts obj is '2.0' | undefined {
+  if (typeof obj !== 'undefined' && obj !== '2.0') {
+    throw new Error('Must be JSONRPC 2.0');
+  }
+}
+
+export function parseTRPCMessage(
```
Comment (juliusmarminge) [inline packages/server/src/rpc/parseTRPCMessage.ts:49]: Add a jsdoc comment thst this is public 

`/** @public */`

**Comment by Guria** [comment]
updated: returned lost line and add public jsdoc

**Review by juliusmarminge — APPROVED** [review]

**Comment by Guria** [comment]
@KATT  Anything else I need to do here, or just wait?

**Comment by KATT** [comment]
Thanks for the nudge, I will look soon!

**Review by KATT — APPROVED** [review]

**Pull request merged**